src/item.cpp

```diff
@@ -6297,7 +6297,7 @@ void item::fill_with( item &liquid, long amount )
             return;
         }
         ammo_set( liquid.typeId(), ammo_remaining() + amount );
-    } else if( !is_container_empty() ) {
+    } else if( is_food_container() ) {
```

Comment:
Removing the check for non-empty container will try to calculate the temperature even if there is no drinkable liquid.

Reply:
`is_food_container()` returns false if the container is empty.